tests/rados/test_cephdf.py

```diff
@@ -286,3 +293,185 @@ def run(ceph_cluster, **kw):
+                empty_devices = rados_obj.get_available_devices(
+                    node_name=node_obj.hostname, device_type="hdd"
+                )
+                if len(empty_devices) < 1:
+                    log.error(
+                        f"Need at least 1 spare disks available on host {node_obj.hostname}"
+                    )
+                    raise Exception(
+                        f"One spare disk not available on host {node_obj.hostname}"
+                    )
+
+                # for dev in empty_devices:
+                lvm_list[node].append(
+                    create_lvms(
+                        node=node_obj, count=1, size="10G", devices=empty_devices[0]
+                    )
+                )
+                log.info(
+                    f"List of LVMs created on {node_obj.hostname}: {lvm_list[node]}"
+                )
+
+            # add backup hosts to the cluster and deploy OSDs
+            service_obj.add_new_hosts(add_nodes=["node12", "node13"], deploy_osd=False)
+
+            # deploy single OSD on lvm devices created
+            for node in ["node12", "node13"]:
+                node_obj = get_node_by_id(ceph_cluster, node)
+                log.info(f"Proceeding to deploy OSDs on {node_obj.hostname}")
+                add_cmd = f"ceph orch daemon add osd {node_obj.hostname}:data_devices={lvm_list[node][0]}"
+                out, err = cephadm.shell(args=[add_cmd])
+
+                if not (
+                    "Created osd" in out and f"on host '{node_obj.hostname}" in out
+                ):
+                    log.error(f"OSD addition on {node_obj.hostname} failed")
+                    log.error(err)
+                    raise Exception(f"OSD addition on {node_obj.hostname} failed")
+
+            _pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            log.info(f"{pool_name} pool stat: {_pool_stat}")
+
+            time.sleep(30)
+            # execute max_avail check across the cluster
+            if not rados_obj.verify_max_avail():
+                log.error("MAX_AVAIL deviates on the cluster more than expected")
+            log.info("MAX_AVAIL on the cluster are as per expectation")
+
+            # expand the OSD LVMs on backup node12
+            node12_obj = get_node_by_id(ceph_cluster, "node12")
+            for lvm in lvm_list["node12"]:
+                _cmd = f"lvextend -L 15G {lvm}"
+                node12_obj.exec_command(cmd=_cmd, sudo=True)
+
+            log.info("LVM size on node12 increased from 10G to 15G")
+            node12_osds = rados_obj.collect_osd_daemon_ids(osd_node=node12_obj)
+
+            # for each OSD on node 12, execute bluefs-bdev-expand
+            for osd_id in node12_osds:
+                out = bluestore_obj.block_device_expand(osd_id=osd_id)
+                log.info(out)
+                assert "device size" in out and "Expanding" in out
+            log.info(f"OSDs {node12_osds} should now be 15G each")
+
```

Comment:
Check osd metadata if the increased size is reported correctly post bdev expand workflow.

Reply:
Added code to verify bluestore_bdev_size from OSD metadata